### PR TITLE
feat/MET-2416-Redirection-Ids

### PR DESCRIPTION
**Pull request by andyjmaclean**
- this has not to be merged until it has been tested against the backend (pending)
- development / review requires that the mock data service be running

### CODE REVIEW

File: src/app/dataset/datasetform/datasetform.component.spec.ts [hunk 1]
```diff
@@ -71,4 +70,50 @@ describe('DatasetformComponent', () => {
...
+    component.addRedirectionId(existingId);
+    expect(component.datasetForm.dirty).toBeFalsy();
+    component.addRedirectionId(newId);
+    expect(component.datasetForm.dirty).toBeTruthy();
+  });
+
+  it('should remove redirection ids', () => {
+    fixture.detectChanges();
+    const existingId = '123';
+    const newId = 'some_id';
```
Comment (mirjamverloop) [inline src/app/dataset/datasetform/datasetform.component.spec.ts]: As these are repeated (here and in the previous scenario), could you move them out of the "it" and define them just once?

File: src/app/dataset/datasetform/datasetform.component.ts [hunk 2]
```diff
@@ -85,12 +85,45 @@ export class DatasetformComponent implements OnInit {
...
+  */
+  removeRedirectionId(val: string): void {
+    const ids = this.redirectionIds;
+    const existingIndex = ids.value.findIndex((id: string) => id === val);
+    if (existingIndex > -1) {
+      ids.removeAt(existingIndex);
+      this.datasetForm.markAsDirty();
+    }
+  }
+
```
Comment (mirjamverloop) [inline src/app/dataset/datasetform/datasetform.component.ts]: As these two methods are almost identical, can you somehow combine the code? Eg, where you pass in if it should be removed or added?

File: src/app/dataset/redirection/redirection.component.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  /* event handler for keystrokes on the new redirection field
+  /* @param {KeyboardEvent} e - the key event
+  */
+  onKeyupRedirect(e: KeyboardEvent): void {
+    if (this.newIdString && this.newIdString.length > 0) {
+      if (this.ignoredKeys.indexOf(e.which) > -1) {
+        return;
+      }
+      if (e.shiftKey || e.ctrlKey || e.altKey) {
+        console.log('skip modifier');
```
Comment (mirjamverloop) [inline src/app/dataset/redirection/redirection.component.ts]: Remove the console.log.

File: src/app/_translate/lang-en.ts [hunk 4]
```diff
@@ -127,6 +129,7 @@ export const LANG_EN_TRANS = {
   provider: 'Provider',
   rawxml: 'Raw XML',
   recordxpath: 'Record XPath',
+  redirectsTo: 'Redirects to',
   refresh: 'Refresh',
   register: 'Register to Metis',
   registererror: "Your account doesn't exist or is not ready yet",
```
[the remark this answers is not in the document]
Comment (andyjmaclean) [reply src/app/_translate/lang-en.ts]: This has been the case for as long as I'm aware (when I add a new one I try to follow the convention used for similar keys).  To standardise all of them would impact a lot of html files (nearly all of them) so this should probably be a separate (technical debt) ticket

File: src/app/dataset/datasetform/datasetform.component.spec.ts (hunk 1 above)
Comment (andyjmaclean) [reply src/app/dataset/datasetform/datasetform.component.spec.ts]: Done

File: src/app/dataset/datasetform/datasetform.component.ts (hunk 2 above)
Comment (andyjmaclean) [reply src/app/dataset/datasetform/datasetform.component.ts]: Done

File: src/app/dataset/redirection/redirection.component.ts (hunk 3 above)
Comment (andyjmaclean) [reply src/app/dataset/redirection/redirection.component.ts]: Done

File: src/app/_translate/lang-en.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (mirjamverloop) [reply src/app/_translate/lang-en.ts]: camelCase maybe, for readability?

**Pull request merged**